Extension/src/LanguageServer/extension.ts

```diff
@@ -249,27 +248,42 @@ async function installVsix(vsixLocation: string, updateChannel: string): Promise
                     'Resources', 'app', 'bin', 'code') + '"';
             } else {
                 const vsCodeBinName: string = path.basename(process.execPath);
-                const stdout: Buffer = execSync('which ' + vsCodeBinName);
-                return stdout.toString().trim();
+                try {
+                    const stdout: Buffer = execSync('which ' + vsCodeBinName);
+                    return stdout.toString().trim();
+                } catch (error) {
+                    return undefined;
+                }
             }
         }(platformInfo);
-        if (!vsCodeScriptPath) {
-            return Promise.reject(new Error('Failed to find VS Code script'));
-        }
 
         // Install the VSIX
-        const installCommand: string = vsCodeScriptPath + ' --install-extension ' + vsixLocation;
-        try {
-            if (updateChannel === 'Default') {
-                // Uninstall the current version, as the version to install is a previous version
-                const uninstallCommand: string = vsCodeScriptPath + ' --uninstall-extension ms-vscode.cpptools';
-                execSync(uninstallCommand);
+        return new Promise<void>((resolve, reject) => {
+            let process: ChildProcess = spawn(vsCodeScriptPath, ['--install-extension', vsixLocation]);
+            if (process.pid === undefined) {
+                reject(new Error('Failed to launch VS Code script process for installation'));
+                return;
             }
-            execSync(installCommand);
-            return Promise.resolve();
-        } catch (error) {
-            return Promise.reject(new Error('Failed to install VSIX'));
-        }
+
+            // Timeout the process if no response is sent back. Ensures this Promise resolves/rejects
+            const timer: NodeJS.Timer = setTimeout(() => {
+                process.kill();
+                reject(new Error('Failed to receive response from VS Code script process for installation within 1s.'));
+            }, 1000);
```

Comment:
I think you should allow a few more seconds to this timer.  There's no real rush to kill the process.  How about 10-15 seconds?

Reply:
Sure!